lib/components/Drawer/Drawer.test.tsx

```diff
@@ -0,0 +1,190 @@
+  it('should not be visible until open', () => {
+    const { getByTestId, queryByRole } = renderDrawer();
+
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+
+    userEvent.click(getByTestId('buttonBefore'));
+
+    expect(queryByRole('dialog')).toBeVisible();
+  });
+
+  it('should have the correct aria properties when open', () => {
+    const { getByTestId, queryByRole, getByLabelText } = renderDrawer();
+
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+
+    userEvent.click(getByTestId('buttonBefore'));
+
+    const dialog = queryByRole('dialog');
+    expect(dialog?.getAttribute('aria-modal')).toBe('true');
+    expect(getByLabelText(DRAWER_TITLE)).toBe(dialog);
+  });
+
+  it('should focus the title element', () => {
+    const { getByTestId, getByText } = renderDrawer();
+
+    userEvent.click(getByTestId('buttonBefore'));
+
+    expect(getByText(DRAWER_TITLE)).toHaveFocus();
+  });
+
+  it('should trap focus in the dialog', () => {
+    const {
+      getByTestId,
+      getByRole,
+      getByText,
+      getByLabelText,
+    } = renderDrawer();
+
+    const dialogOpenButton = getByTestId('buttonBefore');
+    userEvent.tab();
+    expect(dialogOpenButton).toHaveFocus();
+
+    userEvent.click(dialogOpenButton);
+    const buttonInside = getByTestId('buttonInside');
+    const closeButton = getByLabelText(CLOSE_BUTTON_LABEL);
+    const dialogTitle = getByText(DRAWER_TITLE);
+
+    expect(dialogTitle).toHaveFocus();
+
+    userEvent.tab({ focusTrap: getByRole('dialog') });
+    expect(buttonInside).toHaveFocus();
+
+    userEvent.tab({ focusTrap: getByRole('dialog') });
+    expect(closeButton).toHaveFocus();
+
+    userEvent.tab({ focusTrap: getByRole('dialog') });
+    expect(buttonInside).toHaveFocus();
+
+    userEvent.tab({ shift: true, focusTrap: getByRole('dialog') });
+    expect(closeButton).toHaveFocus();
+
+    userEvent.tab({ shift: true, focusTrap: getByRole('dialog') });
+    expect(buttonInside).toHaveFocus();
+  });
+
+  it('should close when hitting escape', async () => {
+    const { getByTestId, queryByRole, getByRole } = renderDrawer();
+
+    const dialogOpenButton = getByTestId('buttonBefore');
+    userEvent.click(dialogOpenButton);
+    fireEvent.keyDown(getByRole('dialog'), { keyCode: ESCAPE });
+
+    await waitForElementToBeRemoved(() => queryByRole('dialog'), {
+      timeout: EXIT_TIMEOUT,
+    });
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should hide document content outside of dialog from screen readers when open', async () => {
+    const { getByTestId, queryAllByRole, queryByRole } = renderDrawer();
+
+    expect(queryAllByRole('textbox').length).toBe(2);
+
+    const dialogOpenButton = getByTestId('buttonBefore');
+    userEvent.click(dialogOpenButton);
+
+    await waitFor(() => queryByRole('dialog'));
+    await new Promise((resolve) => setTimeout(resolve, 500)); // Wait for transition
+
+    expect(queryAllByRole('textbox').length).toBe(0);
```

Comment:
As per previous comment:
```suggestion
    await waitFor(() => expect(queryAllByRole('textbox').length).toBe(0))
```